global/matlab/mixtureModel.m

```diff
@@ -77,17 +84,26 @@ function mixtureModel(inFile,outFile,wbcmd,melocmd)
 [stat(4),~] = system(sprintf(...
   '%s -i %s --ICs=%s --mix=%s/grot -o %s --Oall --report -v --mmthresh=0',... 
   melocmd,inFile, inFile, tDir, tDir),'-echo');
-[stat(5),~] = system(sprintf(...
-    'FSLOUTPUTTYPE=%s;fslmerge -t %s $(ls %s/stats/thresh_zstat* | sort -V);FSLOUTPUTTYPE=%s;',...
-    FSLOUTPUTTYPE, outFile, tDir ,FSLOUTPUTTYPE0),'-echo');
-[stat(6),~] = system(['rm -r ' tDir],'-echo');% clean up temporary files
+% [stat(5),~] = system(sprintf(...
+%     'FSLOUTPUTTYPE=%s;fslmerge -t %s $(ls %s/stats/thresh_zstat* | sort -V);FSLOUTPUTTYPE=%s;',...
+%     FSLOUTPUTTYPE, outFile, tDir ,FSLOUTPUTTYPE0),'-echo');% assumes every output has one map
+[stat(5),mapCounts] = system(sprintf(...
+  "find %s/stats/ -name 'thresh_zstat*' -exec sh -c 'fslval $0 dim4' {} \\;",tDir));
```

Comment:
`sh -c`?  $0?  That is not good practice ($0 is what `ps`/`top`/`htop` show as the process name).  See if this works.

```suggestion
  "find %s/stats/ -name 'thresh_zstat*' -exec fslval {} dim4 \\;",tDir));
```

Reply:
If `find -exec` ignores `$PATH` and can't find `fslval`, we can at least avoid legacy `sh` and abusing $0 for arguments:

```suggestion
  "find %s/stats/ -name 'thresh_zstat*' -exec bash -c 'fslval ""$1"" dim4' bash {} \\;",tDir));
```